### ISSUE TITLE
issue: Root error messages returned from schema are immediately removed during form submit

### ISSUE THREAD

**Issue by solomonhawk** [react-hook-form/resolvers#561]
### Version Number

7.43.9

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-zod-root-error-je3yve?file=/src/App.tsx

### Discord Question Thread link

https://discord.com/channels/754891658327359538/1108113385968644217/1108113385968644217

### Steps to reproduce

This issue relates to and is observed when using `zod@3.21.4` and `@hookform/resolvers@3.1.0` and using a zod schema resolver with `useForm`.

1. Go to ['the sandbox'](https://codesandbox.io/s/react-hook-form-zod-root-error-je3yve?file=/src/App.tsx)
2. Read the summary/instructions therein, or keep reading here
4. Open the Console inside the sandbox
5. Click on "Submit"
6. Despite the form being invalid, it is submitted (the submit handler is called with `{}` which is invalid data)
7. Find the place in the schema where the path of the custom global error is specified and change it from `["root", "global"]` to just `["global"]`
8. Submit the form again, it should fail validation and display the global error message

## Longer explanation of the problem
**Copied from within the sandbox:**

This example demonstrates an issue with assigning errors to "root.<field>" (global form errors) from within a zod schema. There are 2 different issues that can be observed.

1. If the schema contains _only_ errors on "root.<field>" keys, then the form is considered valid (it should be invalid). You can observe this by submitting the form - the `onSubmit` callback is called but with an empty object, which should not be valid.

2. Errors on "root.<field>" are removed from the form state during submission, so they cannot be seen in the UI. This happens [here](https://github.com/react-hook-form/react-hook-form/blob/v7.43.9/src/logic/createFormControl.ts#L1038-L1058) during submission but _after_ the schema is executed which means any root errors added by the schema _for this submission_ are immediately removed.

If you uncomment L41 and comment out L44, so that the schema appends an error to "global" instead of "root.global", you will observe that this error 1) prevents the submission from being considered valid and 2) displays in the UI because it isn't stripped during submission.

If you click the "Set Error" button, you can see that manually calling `setError` with a "root.<field>" key does insert the error properly _outside of submit_.

### Expected behaviour

Error messages under the `"root.<field>"` namespace returned from the provided schema during submission should exist in the form error state _at least until the next submission_.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Suggested Fix

_I'm not super familiar with the codebase, so take this with a grain of salt..._

Move `unset(_formState.errors, 'root')` above where we call `const { errors, values } = await _executeSchema();`. ([related code](https://github.com/react-hook-form/react-hook-form/blob/v7.43.9/src/logic/createFormControl.ts#L1038-L1058))

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Amanjoshi1111** [comment]
In my project I have a field array name "items" (in which I have validation check that it can't be empty).
I am using zod validator latest version.

When I am not adding any item in "items" array it's showing error  message and the message still persist after submitting  and it's present in errors.items.message (which is expected) 

but when I am adding some entries and removing them all and then submitting it's now  not showing error message because this time the error  message is present in errors.root.items.message (IDK why)

TLDR: In some cases RHF errors object will showing message inside errors.root.items.message instead of error.items.message, which don't show validation error in UI.

**Comment by information-security** [comment]
> In my project I have a field array name "items" (in which I have validation check that it can't be empty). I am using zod validator latest version.
> 
> When I am not adding any item in "items" array it's showing error message and the message still persist after submitting and it's present in errors.items.message (which is expected)
> 
> but when I am adding some entries and removing them all and then submitting it's now not showing error message because this time the error message is present in errors.root.items.message (IDK why)
> 
> TLDR: In some cases RHF errors object will showing message inside errors.root.items.message instead of error.items.message, which don't show validation error in UI.

@jorisre 
I am also facing the exact same behavior in the latest zod (3.24.3) and resolvers (5.0.1) versions. I think this issue needs to be reopened.

**Comment by Jazzmanpw** [comment]
I wonder why they closed this issue. [The line that unsets the root error](https://github.com/react-hook-form/react-hook-form/blob/60a7f5a3080ba1d7c171fb9c2fc3bf8f2c24f176/src/logic/createFormControl.ts#L1247) is still in master, and I couldn't find any commits that edited the file and mentioned root error handling (unless I didn't recognize them). And I still face the issue with setting a root error using a zod schema

Any chances you'll get back to this issue and find a way to fix it/explain why you can't fix it?
